cwms-data-api/src/main/java/cwms/cda/data/dto/basin/Basin.java

```diff
@@ -0,0 +1,164 @@
+
+    public Double getContributingArea()
+    {
+        return contributingArea;
+    }
+
+    public LocationIdentifier getParentBasinId()
+    {
+        return parentBasinId;
+    }
+
+    public String getAreaUnit()
+    {
+        return areaUnit;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        Basin basin = (Basin) o;
+        return Objects.equals(getBasinId(), basin.basinId)
+                && Objects.equals(getPrimaryStreamId(), basin.primaryStreamId)
+                && Objects.equals(getSortOrder(), basin.sortOrder)
+                && Objects.equals(getBasinArea(), basin.basinArea)
+                && Objects.equals(getContributingArea(), basin.contributingArea)
+                && Objects.equals(getParentBasinId(), basin.parentBasinId)
+                && Objects.equals(getAreaUnit(), basin.areaUnit);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getBasinId(), getPrimaryStreamId(), getSortOrder(), getBasinArea(),
+                getContributingArea(), getParentBasinId(), getAreaUnit());
+    }
+
+    public static class Builder
+    {
+        private LocationIdentifier basinId;
+        private Double sortOrder;
+        private Double basinArea;
+        private Double contributingArea;
+        private LocationIdentifier parentBasinId;
+        private String areaUnit;
+        private LocationIdentifier primaryStreamId;
+
+        public Builder withBasinId(LocationIdentifier basinId)
+        {
+            this.basinId = basinId;
+            return this;
+        }
+
+        public Builder withPrimaryStreamId(LocationIdentifier primaryStreamId)
+        {
+            this.primaryStreamId = primaryStreamId;
+            return this;
+        }
+
+        public Builder withSortOrder(Double sortOrder)
+        {
+            this.sortOrder = sortOrder;
+            return this;
+        }
+
+        public Builder withBasinArea(Double basinArea)
+        {
+            this.basinArea = basinArea;
+            return this;
+        }
+
+        public Builder withContributingArea(Double contributingArea)
+        {
+            this.contributingArea = contributingArea;
+            return this;
+        }
+
+        public Builder withParentBasinId(LocationIdentifier parentBasinId)
+        {
+            this.parentBasinId = parentBasinId;
+            return this;
+        }
+
+        public Builder withAreaUnit(String areaUnit)
+        {
+            this.areaUnit = areaUnit;
+            return this;
+        }
+
+        public Basin build()
+        {
+            return new Basin(this);
+        }
+    }
+
+    @Override
+    public void validate() throws FieldException {
+        if (this.basinId == null) {
+            throw new FieldException("Basin identifier field can't be null");
+        }
```

Comment:
add a basinId.validate() call after this

Reply:
Fixed in latest push